src/Unused/Types.hs

```diff
@@ -18,20 +19,40 @@ data TermMatch = TermMatch
     , occurrences :: Int
     } deriving Show
 
-type TermMatchSet = Map.Map String [TermMatch]
+data TermResults = TermResults
+    { matches :: [TermMatch]
+    , totalFiles :: Int
+    , totalOccurrences :: Int
+    }
+
+type TermMatchSet = Map.Map String TermResults
 
 type ParseResponse = Either ParseError TermMatchSet
 
+resultsFromMatches :: [TermMatch] -> TermResults
+resultsFromMatches m =
+    TermResults
+        { matches = m
+        , totalFiles = totalFiles'
+        , totalOccurrences = totalOccurrences'
+        }
+  where
+    totalFiles' = length m
+    totalOccurrences' = sum $ fmap occurrences m
+
 responseFromParse :: Either ParseError [TermMatch] -> ParseResponse
 responseFromParse =
-    fmap $ Map.fromList . groupBy term
+    fmap $ Map.fromList . map (mapSnd resultsFromMatches) . groupBy term
 
 withOneFile :: ParseResponse -> ParseResponse
-withOneFile = fmap $ Map.filterWithKey (\_ a -> length a == 1)
+withOneFile = fmap $ Map.filterWithKey (\_ a -> totalFiles a == 1)
 
 withOneOccurrence :: ParseResponse -> ParseResponse
-withOneOccurrence = fmap $ Map.filterWithKey (\_ a -> (sum $ fmap occurrences a) == 1)
+withOneOccurrence = fmap $ Map.filterWithKey (\_ a -> totalOccurrences a == 1)
+
+listFromMatchSet :: TermMatchSet -> [(String, [TermMatch])]
+listFromMatchSet =
+  map (mapSnd matches) . Map.toList
 
-notMatchingPath :: String -> ParseResponse -> ParseResponse
-notMatchingPath s =
-    fmap $ Map.map $ filter (not . isInfixOf s . path)
+mapSnd :: (a -> b) -> (c, a) -> (c, b)
+mapSnd f (a, b) = (a, f b)
```

Comment:
This is `second` from `Data.Bifunctor`, https://hackage.haskell.org/package/bifunctors-3.2.0.1/docs/Data-Bifunctor.html#v:second